onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -106,7 +110,10 @@ std::vector<std::unique_ptr<GraphTransformer>> GenerateTransformers(TransformerL
     case TransformerLevel::Level1: {
       std::unordered_set<std::string> l1_execution_providers = {};
 
-      transformers.emplace_back(onnxruntime::make_unique<ConstantFolding>(l1_execution_providers));
+      // TODO hack - constant folding currently doesn't work after mixed precision transformation so it's disabled for now
+      //             ORT uses CPU kernels to evaluate constant values but some of them don't support fp16
+      //transformers.emplace_back(onnxruntime::make_unique<ConstantFolding>(l1_execution_providers));
```

Comment:
Sherlock: Fix this - https://msdata.visualstudio.com/Vienna/_boards/board/t/Training%20Runtime/Features/?fullScreen=true&workitem=604684

Reply:
reverted the code, not to include it